lib/portlayer/storage/image_cache.go

```diff
@@ -153,15 +156,35 @@ func (c *NameLookupCache) CreateImageStore(op trace.Operation, storeName string)
 		return nil, err
 	}
 
-	// Check for existence and rehydrate the cache if it exists on disk.
-	_, err = c.GetImageStore(op, storeName)
-	// we expect this not to exist.
+	// GetImageStore Operation is able to be retried...
+	getStore := func() error {
+		// Check for existence and rehydrate the cache if it exists on disk.
+		_, err = c.GetImageStore(op, storeName)
+		return err
+	}
+	// isRetry will reuse the tasks.IsRetryError and will
+	// retry when appropriate
+	isRetry := func(err error) bool {
+		return tasks.IsRetryError(op, err)
+	}
+
+	config := retry.NewBackoffConfig()
+	config.InitialInterval = time.Second * 30
+	config.MaxInterval = time.Minute
+	config.MaxElapsedTime = time.Minute * 3
+
+	// attempt to get the image store
+	err = retry.DoWithConfig(getStore, isRetry, config)
 	if err == nil {
+		// no error means that the image store exists and we can
+		// safely return
 		return nil, os.ErrExist
 	}
-
-	if !os.IsNotExist(err) {
-		op.Warnf("Error getting image store %s: %s", storeName, err)
+	// log and return any error besides NotExist -- if the image store
+	// doesn't exist it will be created below
+	if err != nil && err != os.ErrNotExist {
+		op.Errorf("Error getting image store %s: %s", storeName, err)
+		return nil, err
```

Comment:
If we return here then we will not have the chance to trigger the (sort of) cleanup on line 202. I also seem to recall there being logic to remove inconsistent or incomplete images in either the `GetImageStore` or `CreateImageStore` paths.

Can we confirm that we're not skipping repair logic by returning here and what the behaviour of the portlayer is if we do error out at this point.

Reply:
I've changed it so that it will continue on the following two errors:
- ErrNotExist
- ErrCorruptImageStore
Those two will attempt to be corrected by the rest of the function.